R/bcdc_search.R

```diff
@@ -176,9 +180,17 @@ bcdc_search <- function(..., license_id = NULL,
     }
   })
 
+  # build query by collating the terms and any user supplied facets
+  # if there are no supplied facets (e.g., is_list_empty(facets) == FALSE just use terms)
   query <- paste0(
-    terms, ifelse(nzchar(terms), "+", ""),
-    paste(names(facets), facets, sep = ":", collapse = "+"))
+    terms,
```

Comment:
No I understand much better what you are doing. The other programs pattern you _could_ use here is an `if` `else` so something like 

```r
if (is_list_empty(facets) {
     _create one query_
    } else {
_create the other query_
}
```
So then rather than the two `ifelse` functions, you've just got each query in it's place. 

One note: You don't need to do this pattern `is_list_empty(facets)`. Rather you can just directly use what is returned by that and then use `!` is you want the opposite. I think this ends being a bit easier to have track of when getting your logicals squared away.

Reply:
I implemented this suggested pattern, it is much more readable and avoids the duplicated `is_empty()` call, which is much better. Thanks for the tips. 🙏